.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,29 @@
+name: Pre-commit Checks
+
+on: [pull_request]
```

Comment:
May want to limit this to only pull requests affecting the `tuning` subdirectory

```yml
on:
  pull_request:
    paths:
      - "tuning/*"
```